backend/src/Parser/CanvasV2.fs

```diff
@@ -0,0 +1,179 @@
+  | SynBinding (_, _, _, _, attrs, _, _, pat, _returnInfo, expr, _, _, _) ->
+    let expr = ProgramTypes.Expr.fromSynExpr availableTypes expr
+
+    let attrs = attrs |> List.collect (fun l -> l.Attributes)
+
+    let randomIds () : PT.Handler.ids =
+      { moduleID = gid (); nameID = gid (); modifierID = gid () }
+
+    match attrs with
+    | [] ->
+      let newFn = ProgramTypes.UserFunction.fromSynBinding availableTypes letBinding
+      { m with fns = newFn :: m.fns }
+
+    | [ attr ] ->
+      match attr with
+      | SimpleAttribute ("DB", [ name ]) ->
+        // let newDB = PT.DB(name, randomIds ())
+        // { m with dbs = newDB :: m.dbs }
+        Exception.raiseInternal
+          $"Not currently supporting DBs - probably makes sense to wait until they are based on a type rather than 'columns'"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("HttpHandler", [ method; route ]) ->
+        let newHttpHanlder = PT.Handler.Spec.HTTP(route, method, randomIds ())
+        { m with handlers = (newHttpHanlder, expr) :: m.handlers }
+
+      | SimpleAttribute ("REPL", [ name ]) ->
+        //let newHandler = PT.Handler.Spec.REPL(name, randomIds ())
+        //{ m with handlers = (newHandler, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting REPLs, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("Worker", [ name ]) ->
+        //let newWorker = PT.Handler.Spec.Worker(name, randomIds ())
+        //{ m with handlers = (newWorker, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting Workers, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | SimpleAttribute ("Cron", [ name; interval ]) ->
+        //let newCron = PT.Handler.Spec.Cron(name, interval, randomIds ())
+        //{ m with handlers = (newCron, expr) :: m.handlers }
+        Exception.raiseInternal
+          $"Not currently supporting Crons, as we can't test them well yet"
+          [ "attr", attr ]
+
+      | _ ->
+        Exception.raiseInternal
+          $"Not sure how to handle this attribute"
+          [ "attr", attr ]
+
+    | _ ->
+      Exception.raiseInternal
+        $"Can only currently support 1 attribute [<...>] on a let binding"
+        [ "attrs", attrs ]
+
+
+/// An F# module has a list of declarations, which can be:
+/// - a group of let bindings
+/// - a group of type definitions
+/// - a single expression
+/// - a nested module (not currently supported)
+/// - etc
+///
+/// (The declarations are presented as _groups_ presumably to support recursive definitions)
+///
+/// This builds up a CanvasModule by parsing each declaration:
+/// - type definitions are parsed as user types
+/// - let bindings with no attributes are parsed as user functions
+/// - let bindings with attributes are parsed
+///   as handlers (i.e. with `[<HttpHandler("method", "path")>]`)
+///   or as DBs (i.e. with `[<DB("name")>]`)
```

Comment:
I'm not clear how a let would make sense to model a DB declaration?

Reply:
My thought about DB definitions is that we might want to define a DB for a type defined outside of this module.
e.g. what if I just want a data store of List?

Reply:
Or should DB definitions be restricted to only Record type definitions?

Reply:
I think for now they could. Agree that stuff like a list of data would be a cool addition, and it probably wouldn't be too much to add that.